### PR TITLE
Follow-up modifications for GFS post process

### LINKED ISSUE #2191: GFS job JGLOBAL_ATMOS_PRODUCTS failed at finding master and flux files
### What is wrong?

The new gfs job 'JGLOBAL_ATMOS_UPP' changed master and flux file names as  gfs.tCCz.masterfHHH.grib2 and gfs.tCCz.sfluxfHHH.grb2.  The downstream job 'JGLOBAL_ATMOS_PRODUCTS' failed at looking for master and flux files as gfs.tCCz.master.grb2fHHH and gfs.tCCz.sfluxgrbfHHH.grib2. 

### What should have happened?

The job 'JGLOBAL_ATMOS_PRODUCTS' should generate pgrb2/pgrb2b files.

### What machines are impacted?

All or N/A

### Steps to reproduce

Run job 'JGLOBAL_ATMOS_PRODUCTS'.

### Additional information

none.

### Do you have a proposed solution?

Update job 'JGLOBAL_ATMOS_PRODUCTS' with new master and flux file names.
In https://github.com/NOAA-EMC/global-workflow/blob/develop/scripts/exglobal_atmos_products.sh#L41
Change 
  master.grb2${fhr3}
into
  master${fhr3}.grb2

In https://github.com/NOAA-EMC/global-workflow/blob/develop/scripts/exglobal_atmos_products.sh#L176
Change
   sfluxgrb${fhr3}.grib2
into
  sflux${fhr3}.grb2

### PR DESCRIPTION
**Pull request by WenMeng-NOAA**
# Description

This PR is the follow-up modifications for gfs post process component:

- Use new naming convention for inline post outputs
- Add missing parm files for gfs pgrb2/pgrb2 generation
- Use new naming conventions of master/flux files in gfs pgrb2/pgrb2 generation
Resolves #2191

# Type of change

- Bug fix (fixes something broken)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?

Yes, ran the 'JGLOBAL_ATMOS_PRODUCTS' job on WCOSS2

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Review by aerorahul — COMMENTED** [review]
The changes here looks good.
The archive script should also be updated to ensure that these updated files are in the list of files that should be archived.
For reference, the archive script used is:
https://github.com/NOAA-EMC/global-workflow/blob/develop/ush/hpssarch_gen.sh

**Comment by WenMeng-NOAA** [comment]
> The changes here looks good. The archive script should also be updated to ensure that these updated files are in the list of files that should be archived. For reference, the archive script used is: https://github.com/NOAA-EMC/global-workflow/blob/develop/ush/hpssarch_gen.sh

@aerorahul Good catching. Updated hpssarch_gen.sh.

**Comment by aerorahul** [comment]
> > The changes here looks good. The archive script should also be updated to ensure that these updated files are in the list of files that should be archived. For reference, the archive script used is: https://github.com/NOAA-EMC/global-workflow/blob/develop/ush/hpssarch_gen.sh
> 
> @aerorahul Good catching. Updated hpssarch_gen.sh.

A few more instances where `sfluxgrb` appears in `hpssarch_gen.sh`. Lines 398, 399, 411.

**Comment by WenMeng-NOAA** [comment]
@aerorahul 

> A few more instances where sfluxgrb appears in hpssarch_gen.sh. Lines 398, 399, 411.
Updated.

**Review by KateFriedman-NOAA — APPROVED** [review]
Changes look good.

File: scripts/exglobal_atmos_products.sh [hunk 1]
```diff
@@ -42,7 +42,7 @@ fi
 # Section creating pressure grib2 interpolated products
 
 # Files needed by ${INTERP_ATMOS_MASTERSH}
-MASTER_FILE="${COM_ATMOS_MASTER}/${PREFIX}master.grb2${fhr3}"
+MASTER_FILE="${COM_ATMOS_MASTER}/${PREFIX}master${fhr3}.grb2"
```
Comment (aerorahul) [inline scripts/exglobal_atmos_products.sh]: @WenMeng-NOAA 
The workflow looks for the master file to kick off the products generation job.
Please update [this](https://github.com/NOAA-EMC/global-workflow/blob/0237dba612a18c64eb513b6e186701cf37a1a5ef/workflow/rocoto/gfs_tasks.py#L1018) line to:
```diff
-                                   'history_file_tmpl': f'{self.cdump}.t@Hz.master.grb2#dep#'},
+                                   'history_file_tmpl': f'{self.cdump}.t@Hz.master#dep#.grb2'},
```

The analysis products job trigger also needs to be updated.  Please update [this](https://github.com/NOAA-EMC/global-workflow/blob/0237dba612a18c64eb513b6e186701cf37a1a5ef/workflow/rocoto/gfs_tasks.py#L908) line to:
```diff
-        data = f'{atm_master_path}/{self.cdump}.t@Hz.master.grb2anl'
+        data = f'{atm_master_path}/{self.cdump}.t@Hz.masteranl.grb2'
```

Finally, the offline UPP for analysis needs to write out the updated filename to `COM`.  Please update [these](https://github.com/NOAA-EMC/global-workflow/blob/0237dba612a18c64eb513b6e186701cf37a1a5ef/parm/post/upp.yaml#L26-L27) lines with:
```diff
-      - ["{{ DATA }}/GFSPRS.GrbF00", "{{ COM_ATMOS_MASTER }}/{{ RUN }}.t{{ current_cycle | strftime('%H') }}z.master.grb2anl"]
-      - ["{{ DATA }}/GFSPRS.GrbF00.idx", "{{ COM_ATMOS_MASTER }}/{{ RUN }}.t{{ current_cycle | strftime('%H') }}z.master.grb2ianl"]
+      - ["{{ DATA }}/GFSPRS.GrbF00", "{{ COM_ATMOS_MASTER }}/{{ RUN }}.t{{ current_cycle | strftime('%H') }}z.masteranl.grb2"]
+      - ["{{ DATA }}/GFSPRS.GrbF00.idx", "{{ COM_ATMOS_MASTER }}/{{ RUN }}.t{{ current_cycle | strftime('%H') }}z.masteranl.grb2.idx"]
```

**Review by aerorahul — CHANGES_REQUESTED** [review]
Need some more updates for jobs to be triggered off the new names.
Also, analysis UPP needs to create the updated filename.

**Comment by aerorahul** [comment]
@WenMeng-NOAA 
Similar workflow updates will be needed in [gefs_tasks.py](https://github.com/NOAA-EMC/global-workflow/blob/develop/workflow/rocoto/gefs_tasks.py)

File: scripts/exglobal_atmos_products.sh (hunk 1 above)
Comment (aerorahul) [reply scripts/exglobal_atmos_products.sh]: Note more updates will be needed in the `hpssarch_gen.sh` for these changes in analysis filenames.

Comment (WenMeng-NOAA) [reply scripts/exglobal_atmos_products.sh]: I will work on that.

Comment (aerorahul) [reply scripts/exglobal_atmos_products.sh]: @WenMeng-NOAA 
I just looked at the EE2 implementation standards and the correct extension for grib2 files is `grib2`.  We should replace `grb2` with `grib2`.

Comment (WenMeng-NOAA) [reply scripts/exglobal_atmos_products.sh]: @aerorahul Just confirm with you that updates will  be in all master/flux/goes/pgrb2/pgrb2b.

Comment (WenMeng-NOAA) [reply scripts/exglobal_atmos_products.sh]: @aerorahul Updated.

File: ush/forecast_postdet.sh [hunk 2]
```diff
@@ -479,8 +479,8 @@ EOF
       ${NLN} "${COM_ATMOS_HISTORY}/${RUN}.t${cyc}z.sfcf${FH3}.nc" "sfcf${FH3}.nc"
       ${NLN} "${COM_ATMOS_HISTORY}/${RUN}.t${cyc}z.atm.logf${FH3}.txt" "log.atm.f${FH3}"
       if [[ ${WRITE_DOPOST} = ".true." ]]; then
-        ${NLN} "${COM_ATMOS_MASTER}/${RUN}.t${cyc}z.master.grb2f${FH3}" "GFSPRS.GrbF${FH2}"
-        ${NLN} "${COM_ATMOS_MASTER}/${RUN}.t${cyc}z.sfluxgrbf${FH3}.grib2" "GFSFLX.GrbF${FH2}"
+        ${NLN} "${COM_ATMOS_MASTER}/${RUN}.t${cyc}z.masterf${FH3}.grb2" "GFSPRS.GrbF${FH2}"
+        ${NLN} "${COM_ATMOS_MASTER}/${RUN}.t${cyc}z.sfluxf${FH3}.grb2" "GFSFLX.GrbF${FH2}"
```
Comment (aerorahul) [inline ush/forecast_postdet.sh]:
```suggestion
        ${NLN} "${COM_ATMOS_MASTER}/${RUN}.t${cyc}z.masterf${FH3}.grib2" "GFSPRS.GrbF${FH2}"
        ${NLN} "${COM_ATMOS_MASTER}/${RUN}.t${cyc}z.sfluxf${FH3}.grib2" "GFSFLX.GrbF${FH2}"
```

File: ush/hpssarch_gen.sh [hunk 3]
```diff
@@ -168,8 +168,8 @@ if [[ ${type} = "gfs" ]]; then
     fhr=$(printf %03i "${fh}")
     if [[ ${ARCH_GAUSSIAN} = "YES" ]]; then
       {
-        echo "${COM_ATMOS_MASTER/${ROTDIR}\//}/${head}sfluxgrbf${fhr}.grib2"
-        echo "${COM_ATMOS_MASTER/${ROTDIR}\//}/${head}sfluxgrbf${fhr}.grib2.idx"
+        echo "${COM_ATMOS_MASTER/${ROTDIR}\//}/${head}sfluxf${fhr}.grb2"
+        echo "${COM_ATMOS_MASTER/${ROTDIR}\//}/${head}sfluxf${fhr}.grb2.idx"
```
Comment (aerorahul) [inline ush/hpssarch_gen.sh]:
```suggestion
        echo "${COM_ATMOS_MASTER/${ROTDIR}\//}/${head}sfluxf${fhr}.grib2"
        echo "${COM_ATMOS_MASTER/${ROTDIR}\//}/${head}sfluxf${fhr}.grib2.idx"
```

File: ush/hpssarch_gen.sh [hunk 4]
```diff
@@ -395,8 +395,8 @@ if [[ ${type} == "gdas" ]]; then
     fh=0
     while [[ ${fh} -le 9 ]]; do
       fhr=$(printf %03i "${fh}")
-      echo "${COM_ATMOS_MASTER/${ROTDIR}\//}/${head}sfluxgrbf${fhr}.grib2"
-      echo "${COM_ATMOS_MASTER/${ROTDIR}\//}/${head}sfluxgrbf${fhr}.grib2.idx"
+      echo "${COM_ATMOS_MASTER/${ROTDIR}\//}/${head}sfluxf${fhr}.grb2"
+      echo "${COM_ATMOS_MASTER/${ROTDIR}\//}/${head}sfluxf${fhr}.grb2.idx"
```
Comment (aerorahul) [inline ush/hpssarch_gen.sh]:
```suggestion
      echo "${COM_ATMOS_MASTER/${ROTDIR}\//}/${head}sfluxf${fhr}.grib2"
      echo "${COM_ATMOS_MASTER/${ROTDIR}\//}/${head}sfluxf${fhr}.grib2.idx"
```

File: ush/hpssarch_gen.sh [hunk 5]
```diff
@@ -408,7 +408,7 @@ if [[ ${type} == "gdas" ]]; then
     done
     flist="001 002 004 005 007 008"
     for fhr in ${flist}; do
-      file="${COM_ATMOS_MASTER/${ROTDIR}\//}/${head}sfluxgrbf${fhr}.grib2"
+      file="${COM_ATMOS_MASTER/${ROTDIR}\//}/${head}sfluxf${fhr}.grb2"
```
Comment (aerorahul) [inline ush/hpssarch_gen.sh]:
```suggestion
      file="${COM_ATMOS_MASTER/${ROTDIR}\//}/${head}sfluxf${fhr}.grib2"
```

**Review by aerorahul — CHANGES_REQUESTED** [review]
some in `hpssarch_gen.sh` were left to be updated.

File: ush/hpssarch_gen.sh (hunk 5 above)
Comment (WenMeng-NOAA) [reply ush/hpssarch_gen.sh]: @aerorahul Updated.

File: parm/post/upp.yaml [hunk 6]
```diff
@@ -23,8 +23,8 @@ analysis:
       - ["{{ COM_ATMOS_ANALYSIS }}/{{ RUN }}.t{{ current_cycle | strftime('%H') }}z.sfcanl.nc", "{{ DATA }}/{{ flux_filename }}"]
   data_out:
     copy:
-      - ["{{ DATA }}/GFSPRS.GrbF00", "{{ COM_ATMOS_MASTER }}/{{ RUN }}.t{{ current_cycle | strftime('%H') }}z.master.grb2anl"]
-      - ["{{ DATA }}/GFSPRS.GrbF00.idx", "{{ COM_ATMOS_MASTER }}/{{ RUN }}.t{{ current_cycle | strftime('%H') }}z.master.grb2ianl"]
+      - ["{{ DATA }}/GFSPRS.GrbF00", "{{ COM_ATMOS_MASTER }}/{{ RUN }}.t{{ current_cycle | strftime('%H') }}z.masteranl.grib2"]
```
Comment (WalterKolczynski-NOAA) [inline parm/post/upp.yaml]: There should be a period separating the `master`/`sflux` from the `anl`/`${fhr3}` in all these just like the lat-lon files have.

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: parm/post/upp.yaml (hunk 6 above)
Comment (WenMeng-NOAA) [reply parm/post/upp.yaml]: @WalterKolczynski-NOAA Updated.

**Comment by WenMeng-NOAA** [comment]
@aerorahul @WalterKolczynski-NOAA Should all pgrb2/pgrb2b be changed as "gfs.t{CC}z.pgrb2a.?p??.f{FFF}.grib2" and gfs.t{CC}z.pgrb2b.?p??.f{FFF}.grib2"?

**Comment by WalterKolczynski-NOAA** [comment]
> @aerorahul @WalterKolczynski-NOAA Should all pgrb2/pgrb2b be changed as "gfs.t{CC}z.pgrb2a.?p??.f{FFF}.grib2" and gfs.t{CC}z.pgrb2b.?p??.f{FFF}.grib2"?

The target filenames for these are `RUN.atmos.tHHz.pres_a.LpLL.fFFF.grib2` and `RUN.atmos.tHHz.pres_b.LpLL.fFFF.grib2.idx`. Changing these will require a lot of downstream changes in the secondary products (e.g. gempak).

**Comment by WalterKolczynski-NOAA** [comment]
We have a spreadsheet documenting the planned filename changes (still in-progress and subject to change): https://docs.google.com/spreadsheets/d/1SzXQnfoUMF0W82Q906A7h9ZLA-vq59kjDH6EGRJuXDY

**Comment by WenMeng-NOAA** [comment]
> > @aerorahul @WalterKolczynski-NOAA Should all pgrb2/pgrb2b be changed as "gfs.t{CC}z.pgrb2a.?p??.f{FFF}.grib2" and gfs.t{CC}z.pgrb2b.?p??.f{FFF}.grib2"?
> 
> The target filenames for these are `RUN.atmos.tHHz.pres_a.LpLL.fFFF.grib2` and `RUN.atmos.tHHz.pres_b.LpLL.fFFF.grib2.idx`. Changing these will require a lot of downstream changes in the secondary products (e.g. gempak).

@WalterKolczynski-NOAA I can only update file naming convention only in pgrb generation, which is included in my GFS post-processing standalone test. Please let me know your preferred approach.

**Comment by WalterKolczynski-NOAA** [comment]
I've scheduled a brief meeting for 15:00 today to discuss. If that doesn't work, we can pick another time.

**Comment by aerorahul** [comment]
awaiting finalizing filenames confirmation.

**Pull request closed without merging**